### DISCUSSION TITLE
Conditional buildrequires and mock --installdeps

### DISCUSSION THREAD

**Discussion by dletai** [rpm-software-management/mock#1483] [Q&A]
How can I install dependencies for a package built with '--with xxx' flags?
For example, suppose in spec file I have
```
%bcond_with lua
%if %{with lua}
BuildRequires: pkgconfig(lua) >= 5.1.0
%endif

```
Outside mock I usually do `rpmbuild -ba --with lua` 
I can solve it easily by installing the dependencies manually, but it would be nice if I could reuse the same command line parameters for both the installdeps and the actual build of the spec.
Is there an option I can pass to mock?
I have tried using `--with=lua` but I think it only works for building the package, not for installdeps.

Conversely, if I have bcond_without and I want to test building without unnecessary packages, how can I tell mock --installdeps to not pull bcond_without conditional packages?

**Comment by praiskup** [comment]
> I have tried using --with=lua but I think it only works for building the package, not for installdeps.

Thank you for this question.  You are right, `--installdeps` doesn't take `--with/out` into account.  Please submit RFE, or even better - open a pull request!
